src/GapicClientTrait.php

```diff
@@ -633,6 +636,12 @@ private function createCallStack(array $callConstructionOptions)
         $callStack = new CredentialsWrapperMiddleware($callStack, $this->credentialsWrapper);
         $callStack = new FixedHeaderMiddleware($callStack, $fixedHeaders, true);
         $callStack = new RetryMiddleware($callStack, $callConstructionOptions['retrySettings']);
+        if (!empty($callConstructionOptions['autoPopulationSettings'])) {
```

Comment:
Yes, it would be better to just add the middleware like this:

```php
$callstack = new RequestAutoPopulationMiddleware(
    $callstack,
    $callConstructionOptions['autoPopulationSettings']
);
```

Then just have the middleware do a no-op if the settings are empty.

Also, we can ensure `autoPopulationSettings` will always be set (see below).